### ISSUE TITLE
Complex placeholder strings with empty results cause server errors when updating the server data

### ISSUE THREAD

**Issue by IlDucci** [WeblateOrg/weblate#4136]
**Describe the bug**

I've found an error: it is not possible to add more than one empty entries in the replacements string.

`#, replacements:{COLOR-GREY}:""`

Makes any elements called {COLOR-GREY} dissappear. This is a behavior I would like to keep.

Writing this:
`#, replacements:{COLOR-GREY}:"":{COLOR-GARNET}:"":{VARIABLE-01}:99`

Causes a server error when pulling the new changes, and the server considers it is up-to-date.

**To Reproduce**

1. Add a complex replacements section to PO files inside the actual repo.
2. Push the changes (via Git, in my case) to the repo.
3. Pull the repo's changes to the Weblate server.
4. The error "Server error" appears.

**Expected behavior**

Complex replacement strings are read, with both text results as well as empty results.

**Screenshots**

**Server configuration and status**

Version 4.1.1 of Weblate. For more info on the server status, you'll have to wait until the actual server admin names them.

**Weblate deploy checks**

**Exception traceback**

**Additional context**

Related to issue #3140 .

**Comment by nijel** [comment]
It would be great if you could get the exception traceback from the server. I've added tests covering this in 636b2be9e2e86f1c6aba9a6fa09e2f3fa49ec360 and 7c02cee42cda2937eb90c0b22ae92ec2c0e8002b and it seems to work just fine.

**Comment by pleonex** [comment]
I can see errors and warnings from Rollbar:

<details><summary>Server version and status</summary>
<p>

```
$ docker-compose exec weblate weblate list_versions
 * Weblate: 4.1.1
 * Django: 3.0.7
 * siphashc: 1.3
 * Whoosh: 2.7.4
 * translate-toolkit: 3.0.0
 * lxml: 4.5.1
 * Pillow: 7.1.2
 * bleach: 3.1.5
 * python-dateutil: 2.8.1
 * social-auth-core: 3.3.3
 * social-auth-app-django: 3.4.0
 * django-crispy-forms: 1.9.1
 * oauthlib: 3.1.0
 * django-compressor: 2.4
 * djangorestframework: 3.11.0
 * django-filter: 2.3.0
 * django-appconf: 1.0.4
 * user-agents: 2.1
 * filelock: 3.0.12
 * setuptools: 40.8.0
 * jellyfish: 0.8.2
 * openpyxl: 3.0.1
 * celery: 4.4.5
 * kombu: 4.6.10
 * translation-finder: 2.1
 * html2text: 2020.1.16
 * pycairo: 1.16.2
 * pygobject: 3.30.4
 * diff-match-patch: 20181111
 * requests: 2.24.0
 * django-redis: 4.12.1
 * hiredis: 1.0.1
 * sentry_sdk: 0.15.1
 * Cython: 0.29.20
 * misaka: 2.1.1
 * GitPython: 3.1.3
 * borgbackup: 1.1.13
 * pyparsing: 2.4.7
 * Python: 3.7.3
 * Git: 2.20.1
 * psycopg2: 2.8.5
 * psycopg2-binary: 2.8.5
 * phply: 1.2.5
 * chardet: 3.0.4
 * ruamel.yaml: 0.16.10
 * tesserocr: 2.5.1
 * akismet: 1.1
 * boto3: 1.14.6
 * zeep: 3.4.0
 * aeidon: 1.7.0
 * iniparse: 0.5
 * mysqlclient: 1.4.6
 * Mercurial: 5.4.1
 * git-svn: 2.20.1
 * git-review: 1.28.0
 * hub: 2.13.0
 * lab: 0.16
 * Redis server: 4.0.14
 * PostgreSQL server: 9.6.18
 * Database backends: django.db.backends.postgresql
 * Cache backends: default:RedisCache, avatar:FileBasedCache
 * Email setup: django.core.mail.backends.smtp.EmailBackend: smtp.sendgrid.net
 * OS encoding: filesystem=utf-8, default=utf-8
 * Celery: redis://cache:6379/1, redis://cache:6379/1, regular
 * Platform: Linux 3.10.0-1127.13.1.el7.x86_64 (x86_64)

$ docker-compose exec weblate weblate check --deploy
System check identified no issues (3 silenced).
```

</p>
</details>

### Exception

<details><summary>Text stacktrace:</summary>
<p>

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python3.7/dist-packages/django/contrib/auth/decorators.py", line 21, in _wrapped_view
    return view_func(request, *args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/django/views/decorators/http.py", line 40, in inner
    return func(request, *args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/weblate/trans/views/git.py", line 139, in update_project
    return perform_update(request, obj)
  File "/usr/local/lib/python3.7/dist-packages/weblate/trans/views/git.py", line 69, in perform_update
    method=request.GET.get("method"),
  File "/usr/local/lib/python3.7/dist-packages/weblate/trans/views/git.py", line 35, in execute_locked
    result = call(*args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/weblate/trans/models/project.py", line 296, in do_update
    return self.on_repo_components(True, "do_update", request, method=method)
  File "/usr/local/lib/python3.7/dist-packages/weblate/trans/models/project.py", line 277, in on_repo_components
    res = getattr(component, call)(*args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/weblate/trans/models/component.py", line 154, in on_link_wrapper
    return func(self, *args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/weblate/trans/models/component.py", line 1142, in do_update
    self.create_translations(request=request)
  File "/usr/local/lib/python3.7/dist-packages/weblate/trans/models/component.py", line 1550, in create_translations
    force, langs, request, changed_template, from_link
  File "/usr/local/lib/python3.7/dist-packages/weblate/trans/models/component.py", line 1680, in _create_translations
    force, langs, request=request, from_link=True
  File "/usr/local/lib/python3.7/dist-packages/weblate/trans/models/component.py", line 1550, in create_translations
    force, langs, request, changed_template, from_link
  File "/usr/local/lib/python3.7/dist-packages/weblate/trans/models/component.py", line 1641, in _create_translations
    self, lang, code, path, force, request=request
  File "/usr/local/lib/python3.7/dist-packages/weblate/trans/models/translation.py", line 84, in check_sync
    translation.check_sync(force, request=request)
  File "/usr/local/lib/python3.7/dist-packages/weblate/trans/models/translation.py", line 405, in check_sync
    self.sync_unit(dbunits, updated, id_hash, unit, pos)
  File "/usr/local/lib/python3.7/dist-packages/weblate/trans/models/translation.py", line 297, in sync_unit
    newunit.update_from_unit(unit, pos, is_new)
  File "/usr/local/lib/python3.7/dist-packages/weblate/trans/models/unit.py", line 452, in update_from_unit
    state = self.get_unit_state(unit, flags)
  File "/usr/local/lib/python3.7/dist-packages/weblate/trans/models/unit.py", line 363, in get_unit_state
    or (flags is not None and "read-only" in self.get_all_flags(flags))
  File "/usr/local/lib/python3.7/dist-packages/weblate/trans/models/unit.py", line 943, in get_all_flags
    self.translation.all_flags, self.extra_flags, override or self.flags
  File "/usr/local/lib/python3.7/dist-packages/weblate/checks/flags.py", line 80, in __init__
    self.merge(flags)
  File "/usr/local/lib/python3.7/dist-packages/weblate/checks/flags.py", line 92, in merge
    for flag in self.get_items(flags):
  File "/usr/local/lib/python3.7/dist-packages/weblate/checks/flags.py", line 150, in parse
    raise ValueError("Unexpected token: {}, state={}".format(token, state))
ValueError: Unexpected token: {COLOR-GARNET}, state=3
```

</p>
</details>

![image](https://user-images.githubusercontent.com/3107481/86655192-d2e7a080-bfe6-11ea-9b45-92263a953d63.png)

### Warning

![image](https://user-images.githubusercontent.com/3107481/86654438-3e7d3e00-bfe6-11ea-9e55-f34dc54a0784.png)

**Comment by nijel** [comment]
So, the string is different from what you've posted in the initial comment:

```
#, replacements:{COLOR-GREY}::{COLOR-GARNET}::{VARIABLE-01}:99
```

I will test that.

**Comment by IlDucci** [comment]
I can assure you I was adding the quotes, so for some reason, they were getting removed. Will check the fix as soon as 4.2 is released in Docker. Thank you!
